py-polars/polars/lazyframe/frame.py

```diff
@@ -932,6 +937,205 @@ def pipe(
+                F.col(c).count().name.prefix("count:"),
+                F.col(c).null_count().name.prefix("null_count:"),
+            ]
+            # mean
+            mean_expr = (
+                F.col(c).to_physical().mean().cast(dtype)
+                if is_temporal
+                else (F.col(c).mean() if is_numeric or dtype == Boolean else null)
+            )
+
+            # standard deviation, min, max
+            expr_std = F.col(c).std() if is_numeric else null
+            min_expr = F.col(c).min() if not skip_minmax(dtype) else null
+            max_expr = F.col(c).max() if not skip_minmax(dtype) else null
+
+            # percentiles
+            pct_exprs = []
+            for p in quantiles:
+                if is_numeric or is_temporal:
+                    pct_expr = (
+                        F.col(c).to_physical().quantile(p, interpolation).cast(dtype)
+                        if is_temporal
+                        else F.col(c).quantile(p, interpolation)
+                    )
+                    sort_cols.add(c)
+                else:
+                    pct_expr = null
+                pct_exprs.append(pct_expr.alias(f"{p}:{c}"))
+
+            if is_numeric or dtype.is_nested() or dtype in (Null, Boolean):
+                has_numeric_result.add(c)
+
+            # add column expressions (in end-state 'metrics' list order)
+            metric_exprs.extend(
+                [
+                    *count_exprs,
+                    mean_expr.alias(f"mean:{c}"),
+                    expr_std.alias(f"std:{c}"),
+                    min_expr.alias(f"min:{c}"),
+                    *pct_exprs,
+                    max_expr.alias(f"max:{c}"),
+                ]
+            )
+
+        # calculate requested metrics in parallel, then collect the result
+        df_metrics = (
+            (
+                # if more than one quantile, sort the relevant columns to make them O(1)
+                # TODO: drop sort once we have efficient retrieval of multiple quantiles
+                self.with_columns(F.col(c).sort() for c in sort_cols)
+                if sort_cols
+                else self
+            )
+            .select(*metric_exprs)
+            .collect()
+        )
+
+        # reshape wide result
+        n_metrics = len(metrics)
+        column_metrics = [
+            df_metrics.row(0)[(n * n_metrics) : (n + 1) * n_metrics]
+            for n in range(self.width)
+        ]
+        summary = dict(zip(self.columns, column_metrics))
+
+        # cast by column type (numeric/bool -> float), (other -> string)
+        for c in self.columns:
+            summary[c] = [  # type: ignore[assignment]
+                None
+                if (v is None or isinstance(v, dict))
+                else (float(v) if (c in has_numeric_result) else str(v))
+                for v in summary[c]
+            ]
+
+        # return results as a DataFrame
+        df_summary = type(df_metrics)._from_dict(summary)
```

Comment:
Nitpick: Maybe just do `from polars.convert import to_dict` and use it here?

Reply:
Agreed; no need to be quite so 'cute' here ;)